### DISCUSSION TITLE
`session` callback does not have `id` property on `session.user`

### DISCUSSION THREAD

**Discussion by jaredleishman** [nextauthjs/next-auth#3427] [Help] [answered]
### Question 💬

![session-user-id](https://user-images.githubusercontent.com/19354410/145693641-fcea3809-96e0-474e-96c0-15d3c116fa1c.png)
How do I store a user's id in the session? Is that bad practice? Should the `DefaultSession.user` be the same as the `DefaultUser` type?

Documentation reference https://next-auth.js.org/configuration/callbacks#session-callback

### How to reproduce ☕️

Inside the `session` callback, try to assign `session.user.id` to `user.id` (with typescript). You'll receive a warning that id does not exist on `session.user`.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/getting-started/typescript

**Reply by jaredleishman** [reply]
Thank you for the rapid response.
